tg/config.py

```diff
@@ -1,36 +1,58 @@
-import configparser
-import mailcap
-import mimetypes
+"""
+Every parameter (except for DEFAULT_CONFIG) can be
+overwritten by external config file
+"""
 import os
+import platform
+import runpy
 
-DEFAULT_CONFIG = os.path.expanduser("~/.config/tg/tg.conf")
+_os_name = platform.system()
+_darwin = "Darwin"
+_linux = "Linux"
+
+
+DEFAULT_CONFIG = os.path.expanduser("~/.config/tg/conf.py")
 DEFAULT_FILES = os.path.expanduser("~/.cache/tg/")
-max_download_size = "10MB"
-record_cmd = "ffmpeg -f avfoundation -i ':0' -ar 22050 -b:a 32k '{file_path}'"
-long_msg_cmd = "vim -c 'startinsert' {file_path}"
-editor = os.environ.get("EDITOR", "vi")
-
-
-def get_cfg(config=DEFAULT_CONFIG):
-    cfg = configparser.ConfigParser()
-    cfg.read(config)
-    return cfg
-
-
-def save_cfg(cfg, config=DEFAULT_CONFIG):
-    config_dir = os.path.dirname(config)
-    if not os.path.isdir(config_dir):
-        os.makedirs(config_dir)
-    with open(config, "w") as f:
-        cfg.write(f)
-
-
-def get_file_handler(file_name, default=None):
-    mtype, _ = mimetypes.guess_type(file_name)
-    if not mtype:
-        return default
-    caps = mailcap.getcaps()
-    handler, view = mailcap.findmatch(caps, mtype, filename=file_name)
-    if not handler:
-        return None
-    return handler
+LOG_LEVEL = "INFO"
+
+API_ID = "559815"
+API_HASH = "fd121358f59d764c57c55871aa0807ca"
+
+PHONE = None
+ENC_KEY = None
+
+TDLIB_PATH = None
+TDLIB_VERBOSITY = 0
+
+MAX_DOWNLOAD_SIZE = "10MB"
+
+# TODO: check platform
+NOTIFY_CMD = "/usr/local/bin/terminal-notifier -title '{title}' -subtitle '{subtitle}' -message '{msg}' -appIcon '{icon_path}'"
+# TODO: check platform
+if _os_name == _linux:
+    RECORD_CMD = "ffmpeg -f alsa -i default -ar 22050 -b:a 32k '{file_path}'"
+else:
+    RECORD_CMD = (
+        "ffmpeg -f avfoundation -i default -ar 22050 -b:a 32k '{file_path}'"
+    )
+
+# TODO: use mailcap instead of editor
+LONG_MSG_CMD = "vim -c 'startinsert' {file_path}"
+EDITOR = os.environ.get("EDITOR", "vi")
+
+if _os_name == _linux:
+    DEFAULT_OPEN = "xdg-open '{file_path}'"
+else:
+    DEFAULT_OPEN = "open '{file_path}'"
+
+if _os_name == _linux:
+    DEFAULT_COPY = "xclip -selection c"
+else:
+    DEFAULT_COPY = "pbcopy"
+
+
+if os.path.isfile(DEFAULT_CONFIG):
+    env_config_params = runpy.run_path(DEFAULT_CONFIG)
+    for param, value in env_config_params.items():
+        if param.isupper():
+            globals()[param] = value
```

Comment:
I didn't get idea of this lines. It breaks current config.
```
    raise ValueError('You must provide bot_token or phone')
ValueError: You must provide bot_token or phone
```

 current config is all lower case
> notify_cmd = /usr/bin/dunstify --appname "{title}" --raw_icon "{icon_path}" "{title} {msg}"

why are you checking for `isuper`? Also, when storing, do you want to cast to upper case config from file
so `notify_cmd -> NOTIFY_CMD`?

Reply:
also, if you planning any changes in config, don't forget to update README

Reply:
README I will update in another patch. We need to rewrite it completely. And now nobody reads it)

New config isn't compatible with old one anyway.
Do you prefer lower case in config? We can do it, if you want.
I am okay keeping it lower cased in user config, but in out codebase, I find it more readable when it's upper